### PR TITLE
Integration Tests for Boiler

### PR DESCRIPTION
**Pull request by HamletDRC**
## Purpose
Integration tests for Boiler

## Does this introduce a breaking change?
```
[ ] Yes
[x] No
```

## Pull Request Type
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: tests/opc-plc-tests/BoilerTests.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        public BoilerTests() : base(new[] { "--ctb" }) { }
+
+        [TearDown]
+        public void TearDown()
+        {
+            InvokeHeaterMethod("HeaterOn");
+        }
+
+        [TestCase]
+        public void shouldStartTurnedOn()
```
Comment (HamletDRC) [inline tests/opc-plc-tests/BoilerTests.cs:23]: what are the naming conventions for integration test methods?

Comment (algattik) [reply tests/opc-plc-tests/BoilerTests.cs:23]: Usually ObjectOfTest_WhenConditions_ThenExpectation, or ObjectOfTest_Expectation

File: tests/opc-plc-tests/BoilerTests.cs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    /// </summary>
+    [TestFixture]
+    public class BoilerTests : SimulatorTestsBase
+    {
+        public BoilerTests() : base(new[] { "--ctb" }) { }
+
+        [TearDown]
+        public void TearDown()
+        {
+            InvokeHeaterMethod("HeaterOn");
```
[the remark this answers is not in the document]
Comment (HamletDRC) [reply tests/opc-plc-tests/BoilerTests.cs:19]: TearDown is executed after each test method, while OneTimeTearDown is executed after each class: 
https://docs.nunit.org/articles/nunit/writing-tests/attributes/teardown.html

The simulator is stateful, and there is a risk of chained tests where one failing test can leave the header off. 

I think it best to leave this code.

File: tests/opc-plc-tests/BoilerTests.cs [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        {
+            var methodNode = NodeId.Create(methodName, OpcPlc.Namespaces.OpcPlcBoiler, Session.NamespaceUris);
+            Session.Call(GetOpcPlcNodeId("Methods"), methodNode);
+        }
+
+        private BoilerDataType GetBoilerModel()
+        {
+            var nodeId = NodeId.Create(BoilerModel.Variables.Boiler1_BoilerStatus, OpcPlc.Namespaces.OpcPlcBoiler, Session.NamespaceUris);
+            var value = Session.ReadValue(nodeId).Value;
+            return (value as ExtensionObject).Body as BoilerModel.BoilerDataType;
```
[the remark this answers is not in the document]
Comment (HamletDRC) [reply tests/opc-plc-tests/BoilerTests.cs:113]: ok

File: tests/opc-plc-tests/BoilerTests.cs (hunk 1 above)
Comment (HamletDRC) [reply tests/opc-plc-tests/BoilerTests.cs:23]: ok

File: tests/opc-plc-tests/BoilerTests.cs [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+                int pressure = model.Pressure;
+
+                pressure.Should().BeLessThan(previousPressure, "pressure should drop when heater is off");
+                previousPressure = pressure;
+            }
+        }
+
+        private void InvokeHeaterMethod(string methodName)
+        {
+            var methodNode = NodeId.Create(methodName, OpcPlc.Namespaces.OpcPlcBoiler, Session.NamespaceUris);
```
[the remark this answers is not in the document]
Comment (HamletDRC) [reply tests/opc-plc-tests/BoilerTests.cs]: @algattik  can you please check my latest commit and let me know if you still think your comment is relevant? 

In the latest commit there are three helper functions: 
* TurnHeaterOn, TurnHeaterOff, and GetBoilerModel

All of the OPC related API calls are encapsulated into those three methods and there is no trace of OPC API in the test methods themselves. 

I could move the NodeId.Create calls into a SetUp method, but that means sprinkling more OPC API info across the file. IMO, I would leave it as-is (now that I refactored a little further)

File: tests/opc-plc-tests/BoilerTests.cs [hunk 5]
```diff
@@ -0,0 +1,126 @@
+namespace OpcPlc.Tests
+{
+    using BoilerModel;
+    using FluentAssertions;
+    using NUnit.Framework;
+    using Opc.Ua;
+
+    /// <summary>
+    /// Tests for the variables defined in the simulator, such as fast-changing and trended nodes.
```
Comment (algattik) [inline tests/opc-plc-tests/BoilerTests.cs:9]: Text should be changed

**Review by algattik — CHANGES_REQUESTED** [review]

**Pull request merged**